plugins/clipboard/plugin.js

```diff
@@ -146,9 +146,9 @@
 
 			CKEDITOR.dialog.add( 'paste', CKEDITOR.getUrl( this.path + 'dialogs/paste.js' ) );
 
-			// Convert image file (if present) to base64 string for Firefox. Do it as the first
-			// step as the conversion is asynchronous and should hold all further paste processing.
-			if ( CKEDITOR.env.gecko ) {
+			// Convert image file (if present) to base64 string for modern browsers except IE (#4612).
+			// Do it as the first step as the conversion is asynchronous and should hold all further paste processing.
+			if ( !CKEDITOR.env.ie ) {
```

Comment:
IMO it would be more readable to, instead of checking for IE, check for support for custom data types in clipboard (so basically other than `Text` and `URL`) – because that's the reason why IEs don't support it. So I propose:

```suggestion
			if ( CKEDITOR.plugins.clipboard.isCustomDataTypesSupported ) {
```

The result is the same, but the code contains more information now.